wallets/react-web3wallet/README.md

```diff
@@ -0,0 +1,261 @@
+    await signClient.extend({ topic });
+
+    /* new */
+    await wallet.extendSession({ topic });
+```
+
+6. Disconnect from a session
+
+```javascript
+    /* old */
+    await signClient.disconnect({
+        topic,
+        reason: getSdkError("USER_DISCONNECTED"),
+    });
+
+    /* new */
+    await web3wallet.disconnectSession({
+        topic,
+        reason: getSdkError("USER_DISCONNECTED"),
+    });
+```
+
+7. Events
+
+```javascript
+    /* old */
+    signClient.on("session_proposal", handler)
+    signClient.on("session_request", handler)
+
+    /* new */
+    web3wallet.on("session_proposal", handler)
+    web3wallet.on("session_request", handler)
+```
+
+## Migrate from `auth-client` to `Web3Wallet`
+
+1. Initialization
+
+```javascript
+
+    // metadata of your app
+    const metadata = {
+        name: "Demo app",
+        description: "Demo Client as Wallet/Peer",
+        url: "www.walletconnect.com",
+        icons: [],
+    };
+
+    /* old */
+    import { AuthClient } from "@walletconnect/auth-client";
+
+    const authClient = await AuthClient.init({
+        projectId: process.env.NEXT_PUBLIC_PROJECT_ID,
+        metadata,
+    })
+
+    /* new */
+    import { Core } from "@walletconnect/core";
+    import { Web3Wallet } from "@walletconnect/web3wallet";
+
+    const core = new Core({
+        projectId: process.env.NEXT_PUBLIC_PROJECT_ID,
+    })
+
+    const web3wallet = await Web3Wallet.init({
+        core, // <- pass the shared `core` instance
+        metadata: `${metaData}`
+    })
+```
+2. Authenticate with a dapp
+
+```javascript
+    /* old */
+    const iss = `did:pkh:eip155:1:${address}`;
+    authClient.on("auth_request", async (event) => {
+        // format the payload
+        const message = authClient.formatMessage(event.params.cacaoPayload, iss);
+        // prompt the user to sign the message
+        const signature = await wallet.signMessage(message);
+        // respond
+        await authClient.respond(
+            {
+                id: args.id,
+                signature: {
+                    s: signature,
+                    t: "eip191",
+                },
+            },
+            iss,
+        );
+    });
+
+    await authClient.core.pairing.pair({ uri, activatePairing: true });
+
+    /* new */
+    const iss = `did:pkh:eip155:1:${address}`;
+    web3wallet.on("auth_request", async (event) => {
+        // format the payload
+        const message = authClient.formatMessage(event.params.cacaoPayload, iss);
+        // prompt the user to sign the message
+        const signature = await wallet.signMessage(message);
+        // respond
+        await web3wallet.respondAuthRequest(
+            {
+                id: args.id,
+                signature: {
+                    s: signature,
+                    t: "eip191",
+                },
+            },
+            iss,
+        );
+    })
+
+    await core.pairing.pair({ uri: request.uri, activatePairing: true })
```

Comment:
Could be simpler to ask people to call via `web3wallet.core.pairing.pair` as in the old pattern. Avoids having to deal with two instances (`core` and `web3wallet`) side by side.